### PR TITLE
Fix the PasswordHasherFactory usage

### PR DESCRIPTION
**Pull request by bytehead**
Missing fixes for #3257.

### CODE REVIEW

**Review by aschempp — DISMISSED** [review]

**Comment by bytehead** [comment]
Tests fail possibly unrelated:

```
There was 1 failure:

1) Contao\ManagerBundle\Tests\Api\ApplicationTest::testGetsCommandsFromPlugins
Failed asserting that actual size 6 matches expected size 4.
```

**Comment by bytehead** [comment]
And what happened to PHPStan? 

```
 ------ ----------------------------------------------------------------------- 
  Line   src/Security/Authentication/ContaoLoginAuthenticationListener.php      
 ------ ----------------------------------------------------------------------- 
  79     Parameter #1 $user of class                                            
         Symfony\Component\Security\Core\Authentication\Token\UsernamePassword  
         Token constructor expects                                              
         Symfony\Component\Security\Core\User\UserInterface, string given.      
 ------ -----------------------------------------------------------------------
```

```php
class UsernamePasswordToken extends AbstractToken
{
    private $credentials;
    private $firewallName;

    /**
     * @param string|\Stringable|UserInterface $user        The username (like a nickname, email address, etc.) or a UserInterface instance
     * @param mixed                            $credentials
     * @param string[]                         $roles
     *
     * @throws \InvalidArgumentException
     */
    public function __construct($user, $credentials, string $firewallName, array $roles = [])
```

**Comment by fritzmg** [comment]
> Tests fail possibly unrelated:

See https://github.com/contao/contao/pull/3745

**Comment by bytehead** [comment]
PHPStan fails because of https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Security/Core/Authentication/Token/UsernamePasswordToken.php#L27.

**Comment by bytehead** [comment]
PHPStan fixes in #3746

File: core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php [hunk 1]
```diff
@@ -32,14 +32,14 @@ class MakeServicesPublicPass implements CompilerPassInterface
         'security.authentication.trust_resolver',
         'security.firewall.map',
         'security.logout_url_generator',
+        'security.password_hasher_factory',
```
Comment (Toflar) [inline core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: This reveals an issue in the logic of this compiler pass. It shouldn't matter wheter it's a service or an alias. We should use `findDefinition()` which checks both, alias and service ID.

Comment (Toflar) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: And catch the exception to do nothing to keep the existing `->has()` logic.

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: You want to fix that issue in this PR?

Comment (Toflar) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: I can create a separate PR :)

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: I'm fine with fixing it - I just don't know if the scope here is correct?

Comment (aschempp) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: I don't think it is 😬

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: I'll provide a separate PR 👍

Comment (bytehead) [reply core-bundle/src/DependencyInjection/Compiler/MakeServicesPublicPass.php:35]: @Toflar see https://github.com/contao/contao/pull/3759

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead.

**Pull request merged**